### ISSUE TITLE
Code Snippet to install on theme

### ISSUE THREAD

**Issue by iorad** [iorad/knowledgebase#6]
@hoatle on freshdesk they allow any script widget to be installed on their freshdesk theme.

Just like what u did with wordpress can we create a code snippet that a freshdesk user could copy to add "Add Tutorial" button on their site?

Here is some reference
We have a portal customization feature in our Estate plan. Portal customization will allow you to add any script / widget / links / banners on the end user portal. More details can be found in the below links. 
https://support.freshdesk.com/solution/categories/45922/folders/74514/articles/64340-how-to-create-a-freshtheme- 
https://support.freshdesk.com/solution/categories/45922/folders/74514/articles/64344-how-to-use-a-ready-made-freshtheme 

@dan753722 and I were disucssing but I dont think I explained it well.

**Comment by iorad** [comment]
@dan753722 here is the plugin we developed for wordpress
https://github.com/iorad/iorad-editor-wp-plugin

**Comment by hoatle** [comment]
@dan753722 FYI, this is currently provided api from iorad for editor embedding on other sites: https://github.com/iorad/iorad/blob/develop/server-iorad/ioradcom/assets/js/iorad.js

and https://github.com/iorad/iorad-editor-wp-plugin leverages that provided api to integrate it into the workpress editor.

**Comment by iorad** [comment]
@dan753722 also if you want to see how the embedded editor works

http://help.iorad.com/wp-admin/post-new.php
username: admin
password: pippen33

Cilck Add tutorial
![add](https://cloud.githubusercontent.com/assets/5091446/7196137/47a370b2-e49a-11e4-880a-413b3f1d6e20.png)

caputre some slides and click close (on the top left)...you will see the player embed url is already pasted in the WP rich text editor

**Comment by hoatle** [comment]
@iorad I've checked on freshdesk and it's possible to use iorad.js provided api to create "Add Tutorial" button there.

**Comment by iorad** [comment]
@dan753722 is putting together a proof of concept.  Unfortunately its not possible to do it in articles...so he will create it in new ticket view and we will show freshdesk.

**Comment by hoatle** [comment]
Do you mean "new article" as "new topic"?

I checked on both new topic or new ticket, and it all shows that POC works.

![image](https://cloud.githubusercontent.com/assets/83062/7215137/21d08d48-e5f3-11e4-83be-6dad9f0064be.png)

Just paste the following code into the new solution, new ticket template and it should show the buttons to create/ edit tutorials:

```
    <script type="text/javascript" src="//www.iorad.com/server/assets/js/iorad.js"></script>
    <script type="text/javascript">
    (function($) {

      iorad.init(function() {
          var latestTutorialParams;

          //iorad is ready now
          $('#create-new-btn').on('click', function() {
              iorad.createTutorial();
          });


          $('#edit-latest-btn').on('click', function() {
              if (latestTutorialParams) {
                  iorad.editTutorial(latestTutorialParams);
              } else {
                  alert('please create a new tutorial first!');
              }
          });


          iorad.on('editor:close', function(tutorialParams) {
              latestTutorialParams = tutorialParams;

              //print the info
              console.log(tutorialParams);
              console.log(iorad.getPlayerUrl(tutorialParams.uid, tutorialParams.tutorialId, tutorialParams.tutorialTitle));
              console.log(iorad.getEmbeddedPlayerUrl(tutorialParams.uid, tutorialParams.tutorialId, tutorialParams.tutorialTitle));

              //append tutorial player into the page
              $('#player-container').empty().append(iorad.getEmbeddedPlayerUrl(tutorialParams.uid, tutorialParams.tutorialId, tutorialParams.tutorialTitle));
          });
      });

    })(jQuery);
    </script>
    <button type="button" id="create-new-btn">Create New</button>
    <button type="button" id="edit-latest-btn">Edit Latest Created</button>
    <div id="player-container"></div>
```

**Comment by hoatle** [comment]
new ticket POC:

![image](https://cloud.githubusercontent.com/assets/83062/7215148/e52978e0-e5f3-11e4-84f5-04d27800fff6.png)

**Comment by iorad** [comment]
Hey Hoat so are u able to put a add tutorial button on new ticket and new article view page?

**Comment by iorad** [comment]
is the create new the iorad button? so when we click it it opens editor and when we click close it pastes embed url in body?

**Comment by hoatle** [comment]
@iorad yes, currently the POC just shows that iorad button could work on new ticket or new article.

Yes, when we click close, we could make it paste embedded url in the body.

**Comment by iorad** [comment]
@dan let me know if u follow this or if we should have hoat do the proof of concept

**Comment by dan753722** [comment]
Hey @iorad @hoatle , Sorry for replying so late. I should go to investigate my github notification settings.....

I am not sure if that's the right place to put the button stuff? I believe that freshtheme portal view is for end users? 

I am almost finished with freshplug poc though. hitting a 401 error using their endpoint. @iorad I have forwarded you an email.

**Comment by hoatle** [comment]
@dan753722 yes, the screenshots above (POC) just shows that it's possible to add buttons there.

For a good working delivery, it requires design and implementation work.

**Comment by dan753722** [comment]
@hoatle ah... I understand where you come from. iorad.js is awesome by the way.

I still argue that this POC doesn't match our use case. that submit ticket view is for a customer who has issues with his/her product, so the customer goes there and submit a ticket. I believe they do not manage knowledgebase. As knowledgebase really should be taken care of by the customer support agents.

The ticket view I have been talking about is something like this:
![image](https://cloud.githubusercontent.com/assets/5967762/7251999/e37fc404-e873-11e4-8711-9cf43efa3d67.png)

By the way, here is a freshplug early stage POC. A customer support agent can create an iorad tutorial by clicking on one button in the ticket view (one of the two only supported locations for fresh plug at the moment, they are really slow comparing to zendesk...)

![freshplug demo](https://cloud.githubusercontent.com/assets/5967762/7252023/25b6a216-e874-11e4-9272-20579a6e3ca5.gif)

@iorad could you please create a repository for me to version control this freshplug? cheers mate.

Daniel
